components/match2/commons/starcraft_starcraft2/opponent_starcraft.lua

```diff
@@ -0,0 +1,107 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Opponent/Starcraft
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Logic = require('Module:Logic')
+local Opponent = require('Module:Opponent')
+local StarcraftRace = require('Module:Race/Starcraft')
+local Table = require('Module:Table')
+local TypeUtil = require('Module:TypeUtil')
+
+local StarcraftOpponent = Table.deepCopy(Opponent)
+
+StarcraftOpponent.types.Player = TypeUtil.extendStruct(Opponent.types.Player, {
+	race = 'string?',
+})
+
+StarcraftOpponent.types.PartyOpponent = TypeUtil.struct{
+	isArchon = 'boolean',
+	--isSpecialArchon = 'boolean',
+	players = TypeUtil.array(StarcraftOpponent.types.Player),
+	type = TypeUtil.literalUnion(unpack(Opponent.partyTypes)),
+}
+
+StarcraftOpponent.types.Opponent = TypeUtil.union(
+	Opponent.types.TeamOpponent,
+	StarcraftOpponent.types.PartyOpponent,
+	Opponent.types.LiteralOpponent
+)
+
+--[[
+Not supported:
+
+Legacy TeamOpponent ({{TeamOpponent|players=...}})
+TeamOpponent without team template ({{TeamOpponent|name=...|short=...}})
+]]
+function StarcraftOpponent.readOpponentArgs(args)
+	local opponent = Opponent.readOpponentArgs(args)
+	local partySize = Opponent.partySize(opponent.type)
+
+	if partySize == 1 then
+		opponent.players[1].race = StarcraftRace.read(args.race)
+
+	elseif partySize then
+		opponent.isArchon = Logic.readBool(args.isarchon)
+		if opponent.isArchon then
+			local archonRace = StarcraftRace.read(args.race)
+			for _, player in ipairs(opponent.players) do
+				player.race = archonRace
+			end
```

Comment:
Looks to me that this is the same as Solo? If it's always gonna be the same between the two

Reply:
Archon is 2 players playing as 1 (they controll the same units/buildings)

Reply:
I meant the function :D

Reply:
in the archon case you have a 2nd player that also needs to get race set
could do this:
```lua
	if partySize and (partySize == 1 or Logic.readBool(args.isarchon)) then
		local race = StarcraftRace.read(args.race)
		for _, player in ipairs(opponent.players) do
			player.race = race
		end
	elseif partySize then
		for playerIx, player in ipairs(opponent.players) do
			player.race = StarcraftRace.read(args['p' .. playerIx .. 'race'])
		end
	end
```

is that what you mean?

Reply:
Yeah something like that was what I was thinking, but at a second thought I'm not too much of a fan of that if statemenet, so let's drop it.